### PR TITLE
Upgrade the azure-sdk-for-go to v67.1.0

### PR DESCRIPTION
**Pull request by jingai-db**
**What type of PR is this?**
Utility PR to version bump the github.com/Azure/azure-sdk-for-go package to v67.1.0
/kind cleanup

Notice that, this PR picks up the efforts made by https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2762

**What this PR does / why we need it**:

There are some features coming that my team is working on that requires the bump. Doing this as a one off to see if the tests all work before we start working on what we actually need.

Notice that, currently Azure SDK v1 is marked for deprecation; we expect it takes time to migrate to ARM SDK as v2. This just unblocks the ongoing efforts.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by jingai-db** [comment]
@CecileRobertMichon `pull-cluster-api-provider-azure-test` failed for `staticcheck`, as one of the 45 linters. By checking the specific error code https://staticcheck.io/docs/checks#SA1019, it is expected to fail. Any suggestion? Thanks!

**Comment by CecileRobertMichon** [comment]
@jingai-db is there a way to exclude all SDK packages from staticcheck in `.golangci.yml` and make a note in #2670 to revert the exception?

File: .golangci.yml [hunk 1]
```diff
@@ -137,6 +137,9 @@ issues:
         - prealloc
     - path: ^test/
       text: exported (.+) should have comment( \(or a comment on this block\))? or be unexported
+    - linters:
+        - staticcheck
+      text: "SA1019:"
```
Comment (CecileRobertMichon) [inline .golangci.yml:142]: is it possible to make this exclusion more targetted to make sure it only applies to the SDK functions and not everything in the codebase?

Comment (jingai-db) [reply .golangci.yml]: I just enriched the text to exclude by additionally referring to the prefix of SDKv1's pkg (found from https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2899/pull-cluster-api-provider-azure-test/1599842175804248064/build-log.txt). Hope it works as intended. PTAL. Thanks!

**Comment by jingai-db** [comment]
@CecileRobertMichon I have an issue to rebase from my previous forked repo and thereby reopened https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2906. PTAL!

**Pull request closed without merging**